runtime/doc/syntax.txt

```diff
@@ -4761,6 +4761,8 @@ DiffChange	diff mode: Changed line |diff.txt|
 DiffDelete	diff mode: Deleted line |diff.txt|
 							*hl-DiffText*
 DiffText	diff mode: Changed text within a changed line |diff.txt|
+							*hl-EndOfBuffer*
+EndOfBuffer	lines after the last line in the buffer
```

Comment:
Hm. But the end of buffer is not really what the '~'s indicate (there is only one end of the buffer, but many filler lines). What about

      EndOfBuffer        filler '~'s after the end of the buffer